public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourceField/ResourceField.tsx

```diff
@@ -26,9 +25,21 @@ function parseResourceDetails(resourceURI: string) {
   };
 }
 
-const ResourceField: React.FC<AzureQueryEditorFieldProps> = ({ query, datasource, onQueryChange }) => {
+interface ResourceFieldProps extends AzureQueryEditorFieldProps {
```

Comment:
What do we gain by extending AzureQueryEditorFieldProps vs explicitly defining what ResourceFieldProps should be?